### PR TITLE
fix simplifyParen method

### PR DESCRIPTION
**Pull request by builtbywill**

### CODE REVIEW

**Review by tragessere — APPROVED** [review]

File: src/editor/editor-editableMathlist.js [hunk 1]
```diff
@@ -1962,22 +1962,20 @@ function removeParen(list) {
...
-                        if (atoms[i].body[j].type !== 'first') nonGenFracCount++;
-                    }
-                    if (nonGenFracCount === 0 && genFracCount === 1) {
-                        // This is a single frac inside a leftright: remove the leftright
-                        atoms[i] = atoms[i].body[genFracIndex];
+            if (atoms[i].type === 'leftright' && atoms[i].leftDelim === '(' && Array.isArray(atoms[i].body)) {
+                let genFracCount = 0;
+                let genFracIndex = 0;
+                let nonGenFracCount = 0;
+                for (let j = 0; atoms[i].body[j]; j++) {
```
Comment (tragessere) [inline src/editor/editor-editableMathlist.js]: Can `body` be null/undefined? might need to check `atoms[i].body && atoms[i].body[j]`

File: dist/src/editor/editor-editableMathlist.js [hunk 2]
```diff
@@ -1967,7 +1967,7 @@ EditableMathlist.prototype.simplifyParen = function(atoms) {
                     let genFracCount = 0;
                     let genFracIndex = 0;
                     let nonGenFracCount = 0;
-                    for (let j = 0; atoms[i].body; j++) {
+                    for (let j = 0; atoms[i].body[j]; j++) {
```
Comment (tragessere) [inline dist/src/editor/editor-editableMathlist.js:1970]: Can `body` be undefined/null? Might need to check `atoms[i].body && atoms[i].body[j]`

Comment (builtbywill) [reply dist/src/editor/editor-editableMathlist.js:1970]: @tragessere Maybe the check for `Array.isArray(atoms[i].body)` is enough?

Comment (tragessere) [reply dist/src/editor/editor-editableMathlist.js:1970]: Oops, that was just cut off of the change preview. Yeah that should work.

**Review by tragessere — APPROVED** [review]

**Pull request merged**